eppo/src/main/java/cloud/eppo/android/EppoClient.java

```diff
@@ -148,71 +151,167 @@ private EppoValue getTypedAssignment(String subjectKey, String flagKey, SubjectA
     }
 
     public String getAssignment(String subjectKey, String flagKey, SubjectAttributes subjectAttributes) {
-        return this.getStringAssignment(subjectKey, flagKey, subjectAttributes);
+        try {
+            return this.getStringAssignment(subjectKey, flagKey, subjectAttributes);
+        } catch (Exception e) {
+            if (this.isGracefulMode) {
+                Log.i(TAG, "error getting assignment value: " + e.getMessage());
+                return null;
+            }
+            throw e;
```

Comment:
Like we did in Javascript, this could be consolidated into a reusable `logOrRethrow()` method.